pkg/controllers/monitoring-stack/controller.go

```diff
@@ -196,3 +219,35 @@ func (r *reconciler) reconcileObject(ctx context.Context, ms *stack.MonitoringSt
 	logger.Info("Updating stack component")
 	return r.k8sClient.Update(ctx, desired)
 }
+
+func (r *reconciler) createGrafanaDSWatch(ctx context.Context) error {
+	if !r.shouldCreateGrafanaDSWatch {
+		return nil
+	}
+	log := r.logger.WithName("create-grafana-ds-watch")
+	var dataSources grafanav1alpha1.GrafanaDataSourceList
+
+	err := r.k8sClient.List(ctx, &dataSources, client.InNamespace("default"))
+
+	if err != nil {
+		log.V(3).Info("grafana data source CRD is not defined")
+		return err
```

Comment:
I think we should follow same the implementation in @fpetkovski 's PR ... https://github.com/rhobs/monitoring-stack-operator/pull/103/files#diff-606d75b2cccce3c13550d55e0c1b27bb1c1fad0431b4f16d466e41dd4e0c801dR355 


I.E. function should return `requeue (bool)` and `error`

```
func setWatch() (requeue bool, err error){

  if err = listDataSources(); err != nil 
    return true, nil 
  }
  err = setWatch()
  return false, err
}
```

So that the caller (Reconcile func) can ... 

```go
if requeue, err := r.setGrafanaDatasourceWatch(); requeue || err != nil {
   if err {
     return ctrl.Result{}, err
  }
  return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
}
```

Reply:
All the patchers in this `reconcile` function return an `error` , I have followed the same model.

Reply:
This part is different from the rest of the patchers. This isn't a patcher to start with :-) 

More importantly ... 
We need the ability to `requeue`  *without* error if GrafanaDataSource CRD is not created and to log **real** errors when setting up a watch fails.